test/e2e/framework/job/wait.go

```diff
@@ -68,6 +70,22 @@ func WaitForJobComplete(ctx context.Context, c clientset.Interface, ns, jobName
 	})
 }
 
+// WaitForJobSuspend uses c to wait for suspend condition for the Job jobName in namespace ns.
+func WaitForJobSuspend(ctx context.Context, c clientset.Interface, ns, jobName string) {
```

Comment:
When using gomega.Eventually, a failure is attributed to the helper, not the test, and the test cannot provide additional details. Better return an error, as explained in https://www.kubernetes.dev/blog/2023/04/12/e2e-testing-best-practices-reloaded/#polling-and-timeouts ("The E2E framework therefore uses a different approach").

Reply:
Thank you.  That was a useful read for this.